doc/compiler/optimizer/LoopOptimizationSummary.md

```diff
@@ -0,0 +1,114 @@
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# [Loop Optimizations](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/IntroLoopOptimizations.md)
+
+## Loop Canonicalization
+Loop Canonicalization transforms `while` loops to if-guarded `do-while` loops
+with a loop invariant (pre-header) block. It places the loop test at the end
+of the trees for the loop, so that the loop back-edge is almost always a
+backwards branch.
+
+## Loop Versioning and Loop Specialization
+During Loop Versioning, loop invariant null checks, bound checks, div checks,
+checkcasts and inline guards (where the `this` does not change in the loop)
+are eliminated from loops by placing equivalent tests outside the loop. It
+also eliminates bound checks where the range of values of the index inside
+the loop is discovered (need to be a compile time constant) and checks are
+inserted outside the loop so that no bound checks fail inside the loop.
+
+This analysis uses induction variable information found by [Value Propagation](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/ValuePropagation.md).
+Another versioning test that is emitted ensures the loop will not run for a
+very long period of time (in which case the async check inside the loop can
+be removed). This special versioning test depends on the number of trees
+(code) inside the loop.
+
+Loop Specialization replaces loop-invariant expressions that are profiled
+and found to be constants, with the constant value after inserting a test
+outside the loop that compares the value to the constant. Note that this
+cannot be done in the the absence of value profiling infrastructure.
+
+## Loop Unrolling and Peeling
+A majority of loops can be unrolled or peeled with or without the loop driving
+tests done after each iteration (loop) body. Usually a peel is inserted before
+the unrolled loop and the residual iterations to be done (at most u-1 residual
+iterations where u is the unroll factor) are also done using the peeled code.
+Peeling aids [Partial Redundancy Elimination (PRE)](optimizer/GlobalOptimizationsSummary.md)
+as code does not have to be moved; instead commoning of dominated expressions can be done
+(which is far easier than code motion because of problems introduced by exception checks
+in Java).
+
```

Comment:
We had a plan to implement loop peeling but we never actually did it. So the documentation should not be included for it.